server/odc-service/src/main/java/com/oceanbase/odc/service/flow/FlowTaskInstanceLoggerService.java

```diff
@@ -0,0 +1,118 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.flow;
+
+import java.io.File;
+import java.util.Optional;
+
+import org.springframework.stereotype.Service;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.oceanbase.odc.common.json.JsonUtils;
+import com.oceanbase.odc.core.shared.constant.ErrorCodes;
+import com.oceanbase.odc.core.shared.constant.TaskType;
+import com.oceanbase.odc.core.shared.exception.NotFoundException;
+import com.oceanbase.odc.metadb.task.TaskEntity;
+import com.oceanbase.odc.service.common.response.SuccessResponse;
+import com.oceanbase.odc.service.dispatch.DispatchResponse;
+import com.oceanbase.odc.service.dispatch.RequestDispatcher;
+import com.oceanbase.odc.service.dispatch.TaskDispatchChecker;
+import com.oceanbase.odc.service.schedule.LoggerProperty;
+import com.oceanbase.odc.service.task.TaskService;
+import com.oceanbase.odc.service.task.executor.logger.LogUtils;
+import com.oceanbase.odc.service.task.model.ExecutorInfo;
+import com.oceanbase.odc.service.task.model.OdcTaskLogLevel;
+
+import cn.hutool.core.io.FileUtil;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author mayang
+ */
+@Service
+@Slf4j
+public class FlowTaskInstanceLoggerService {
+
+    private final FlowTaskInstanceService flowTaskInstanceService;
+    private final RequestDispatcher requestDispatcher;
+    private final TaskDispatchChecker dispatchChecker;
+    private final TaskService taskService;
+    private final LoggerProperty loggerProperty;
+
+    public FlowTaskInstanceLoggerService(FlowTaskInstanceService flowTaskInstanceService,
+            RequestDispatcher requestDispatcher,
+            TaskDispatchChecker dispatchChecker,
+            TaskService taskService,
+            LoggerProperty loggerProperty) {
+        this.flowTaskInstanceService = flowTaskInstanceService;
+        this.requestDispatcher = requestDispatcher;
+        this.dispatchChecker = dispatchChecker;
+        this.taskService = taskService;
+        this.loggerProperty = loggerProperty;
+    }
+
+    @SneakyThrows
+    public String getLog(OdcTaskLogLevel level, Long flowInstanceId, boolean skipAuth) {
+        Optional<TaskEntity> taskEntityOptional =
+                flowTaskInstanceService.getLogDownloadableTaskEntity(flowInstanceId, skipAuth);
+        if (!taskEntityOptional.isPresent()) {
+            log.warn("get log failed, flowInstanceId: {}, skipAuth: {}", flowInstanceId, skipAuth);
+            return String.format("get log failed, id=%s", flowInstanceId);
```

Comment:
`"Get log failed, the log file may not exists or removed, flowInstanceId" + flowInstanceId`

Reply:
ok